modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Decoder.scala

```diff
@@ -111,18 +114,50 @@ trait StandardDecoderInstances {
   }
 
   implicit final def graphDecodeSet[A](implicit A: Decoder[A]): Decoder[Set[A]] = Decoder.instance { c =>
-    c.downSet.foldLeft(Right(Set.empty): Result[Set[A]]) {
-      case (l @ Left(_), _) => l
-      case (Right(set), el) => A(el).map(set + _)
+    c.cursors match {
+      case Some(cs) => cs.traverse(A.apply)
+      case None     => A(c).map(a => Set(a))
+    }
+  }
+
+  implicit final def graphDecodeVector[A](implicit A: Decoder[A]): Decoder[Vector[A]] = Decoder.instance { c =>
+    // TODO: make this stack safe
+    c.narrow.focus match {
+      case Some(IriNode(rdf.nil)) => Right(Vector.empty)
+      case Some(_: IriOrBNode) =>
+        for {
+          head <- A(c.down(rdf.first))
+          tail <- graphDecodeVector[A].apply(c.down(rdf.rest))
+        } yield head +: tail
+      case Some(_: Literal) =>
+        val errMsg =
+          "Unable to decode value to Vector, expected either rdf:nil or node with rdf:first and rdf:rest, but found literal."
+        Left(DecodingFailure(errMsg, c.history))
+      case None =>
+        Left(DecodingFailure("Unable to decode Vector, expected cursor to be centered on a single Node", c.history))
     }
   }
 
   implicit final def graphDecodeList[A](implicit A: Decoder[A]): Decoder[List[A]] = Decoder.instance { c =>
-    c.downList.traverse(A.apply)
+    // TODO: make this stack safe
```

Comment:
Both implementations are the same. Why not to have in the decoder a map method?
Something like
```scala
val decoderVector: Decoder[Vector[A]] = Decoder[List[A]].map(list => list.toVector)
```

Reply:
because it would do two traversals instead of one (performance).